croptalk/tools.py

```diff
@@ -1,52 +1,111 @@
-import os
-from math import sqrt, cos, sin
 from typing import Optional, Union
 
 import requests
-from dotenv import load_dotenv
 from langchain.tools import tool
+from langchain.chat_models import ChatOpenAI
+# from croptalk.load_data import SOB
+import os
+from langchain_core.output_parsers import StrOutputParser
 
-from croptalk.load_data import SOB
+import pandas as pd
+from sqlalchemy import create_engine, text
+from dotenv import load_dotenv
+from langchain_core.prompts import PromptTemplate
 
 load_dotenv("secrets/.env.secret")
+load_dotenv("secrets/.env.shared")
 
 
-@tool("get_SOB_metrics")
-def get_sob_metrics_for_crop_county(state_abbreviation: str, county_name: str, commodity_name: str,
-                                               insurance_plan_name: str, metric: str) -> str:
-    """
-    This tool is used to query the summary of business data (SOB) to retrieve insurance program metrics by coverage level.
+def initialize_llm(model):
+    return ChatOpenAI(
+        model=model,
+        streaming=True,
+        temperature=0,
+    )
 
-    Those metric can be percentage liability indemnified (pct_liability_indemnified), cost to grower (cost_to_grower),
-    the number of policies sold (policies_sold_count) or the premium per quantity (premium_per_quantity)
 
-    For instance, a user might ask : "What is the percentage of policies indemnified for Pierce county in North Dakota,
-    for sunflowers under the APH program?". The tool will answer this question with the following arguments:
+def create_commodity_tool_chain():
```

Comment:
This looks like a duplication of `commodity_chain` in model_llm... Is there a way to reuse rather than duplicate to simplify maintenance?

Reply:
You're right, I just simplified it